### PR TITLE
hw2 implementation

### PR DESCRIPTION
**Pull request by Pharmazon**

### CODE REVIEW

File: NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/Client.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+/**
+ * @author Alexey Shcheglov
+ * @version dated 01.12.2018
+ */
+
+public class Client {
+
+    public static void main(String[] args) {
+
+        ApplicationContext context = new AnnotationConfigApplicationContext(AppConfig.class);
```
Comment (denis-volnenko) [inline NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/Client.java]: final

File: NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/Client.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        Camera camera = context.getBean("camera", Camera.class);
```
Comment (denis-volnenko) [inline NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/Client.java]: final

File: NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/lifecycle/UnproducableCameraRollBeanFactoryPostProcessor.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * @author Alexey Shcheglov
+ * @version dated 01.12.2018
+ */
+
+@Component
+public class UnproducableCameraRollBeanFactoryPostProcessor implements BeanFactoryPostProcessor {
+
+    public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory) throws BeansException {
+
+        String[] beanDefinitionNames = beanFactory.getBeanDefinitionNames();
```
Comment (denis-volnenko) [inline NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/lifecycle/UnproducableCameraRollBeanFactoryPostProcessor.java]: final

File: NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/lifecycle/UnproducableCameraRollBeanFactoryPostProcessor.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        for (String name: beanDefinitionNames){
+
+            BeanDefinition beanDefinition = beanFactory.getBeanDefinition(name);
+            String className = beanDefinition.getBeanClassName();
+
+            try {
+                Class<?> beanClass = Class.forName(className);
```
Comment (denis-volnenko) [inline NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/lifecycle/UnproducableCameraRollBeanFactoryPostProcessor.java]: final

File: NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/lifecycle/UnproducableCameraRollBeanFactoryPostProcessor.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+                UnproducableCameraRoll annotation = beanClass.getAnnotation(UnproducableCameraRoll.class);
```
Comment (denis-volnenko) [inline NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/lifecycle/UnproducableCameraRollBeanFactoryPostProcessor.java]: final

File: NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/Client.java (hunk 1 above)
Comment (Pharmazon) [reply NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/Client.java]: fixed

File: NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/Client.java (hunk 2 above)
Comment (Pharmazon) [reply NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/Client.java]: fixed

File: NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/lifecycle/UnproducableCameraRollBeanFactoryPostProcessor.java (hunk 3 above)
Comment (Pharmazon) [reply NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/lifecycle/UnproducableCameraRollBeanFactoryPostProcessor.java]: fixed

File: NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/lifecycle/UnproducableCameraRollBeanFactoryPostProcessor.java (hunk 4 above)
Comment (Pharmazon) [reply NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/lifecycle/UnproducableCameraRollBeanFactoryPostProcessor.java]: fixed

File: NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/lifecycle/UnproducableCameraRollBeanFactoryPostProcessor.java (hunk 5 above)
Comment (Pharmazon) [reply NewsPortal/src/main/java/ru/shcheglov/geekbrains/hw/hw2/lifecycle/UnproducableCameraRollBeanFactoryPostProcessor.java]: fixed

**Pull request merged**